File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        String answer = "";
        for(int i = 1; i< food.length; i++){
            for(int j = 0; j< food[i] / 2; j++){
                answer+= i;
            }
        }
        answer += "0";
        
        char[] chars = answer.toCharArray();
        
        for(int z = chars.length - 2; z >= 0; z--){
            answer += chars[z];
        }
        return answer;
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    var answer = '';
    for(let i=0; i<food.length; i++){
        if(food[i]%2!==0 && i!==0){
            food[i]=food[i]-1
        }
    }
    for(let j=1; j<food.length; j++){
        if(food[j]!==0){
           for(let z=0; z<food[j]/2; z++){
               answer+=j
           }
        }
    }
    answer+=0
    
    for(let a= answer.length; a > 0; a--){
        if(answer[a-1] !== "0"){
              answer+= answer[a-1]    
        }
    }
    return answer
}